integration/singularity/id_map.go

```diff
@@ -0,0 +1,86 @@
+package singularity
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	"github.com/filecoin-project/motion/blob"
+)
+
+type idMap struct {
+	dir string
+}
+
+func newIDMap(dir string) *idMap {
+	return &idMap{
+		dir: dir,
+	}
+}
+
+func (im *idMap) path(blobID blob.ID) string {
+	return path.Join(im.dir, blobID.String()+".id")
```

Comment:
Can `im.dir` be a filesystem path (possibly containing system-specific delimiters)?  If yes, that it may be necessary to use `filepath.Join`.